qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -162,17 +162,46 @@ class WebEngineSearch(browsertab.AbstractSearch):
 
     Attributes:
         _flags: The QWebEnginePage.FindFlags of the last search.
+        _flag_wrap: An additional flag indicating whether the last search
+                    used wrapping.
         _pending_searches: How many searches have been started but not called
                            back yet.
+        _nowrap_available: Whether the nowrap functionality is available.
+        _active_match: The 1-based index of the currently active matc
+                       on the page.
+        _total_matches: The total number of search matches on the page.
     """
 
     def __init__(self, tab, parent=None):
         super().__init__(tab, parent)
         self._flags = QWebEnginePage.FindFlags(0)  # type: ignore
+        self._flag_wrap = True
         self._pending_searches = 0
+        # The API necessary to stop wrapping was added in this version
+        if qtutils.version_check("5.14"):
+            self._active_match = 0
+            self._total_matches = 0
+            self._nowrap_available = True
+            self._store_match_data_connected = False
+        else:
+            self._nowrap_available = False
```

Comment:
There are a lot of new attributes here, and some of them are conditionally here sometimes, and sometimes they aren't. Would be nice to find a way to clean them up a bit.

What about writing a class (`_WebEngineSearchWrapHandler` or whatever) to hold all those attributes (`self._flag_wrap` included, probably) and saving that as `self._wrap_handler` or so? Then that could be `None` if not available.

That would also make it possible to move some methods such as `_reset_match_data` to that class instead, which keeps this class a bit simpler.

Reply:
It makes some of the statements more verbose, but yeah, I agree. I was a little shy creating new classes in a codebase I'm still not 100% familiar with.